release/llm_tests/serve/benchmark/benchmark_vllm.py

```diff
@@ -0,0 +1,287 @@
+                raise Exception("Server process terminated unexpectedly")
+
+            # Check if server is responding
+            try:
+                import requests
+
+                response = requests.get("http://localhost:8000/health")
+                if response.status_code == 200:
+                    server_ready = True
+                    ColoredLogger.log_server("Server is ready!")
+            except Exception:
+                time.sleep(1)
+                continue
+
+        if not server_ready:
+            raise TimeoutError("Server failed to start within timeout period")
+
+        # Start benchmark
+        ColoredLogger.log_client("Starting benchmark...")
+        results = run_bm(
+            api_url="http://localhost:8000",
+            api_key="NONE",
+            concurrency=[1, 2, 4, 8, 16, 32],
+            run_time="1m",
+            prompt_tokens=256,
+            max_tokens=64,
+            stream=False,
+            summary_file="./results.csv",
+        )
+        print(
+            "Writing final result to AWS Firehose:",
+            json.dumps(results, indent=4, sort_keys=True),
+            sep="\n",
+        )
+
+        ColoredLogger.log_client("Benchmark completed successfully")
+
+    except Exception as e:
+        ColoredLogger.log_error(f"Error during benchmark: {str(e)}")
+        raise
+
+    finally:
+        # Clean up
+
+        if "server_process" in locals():
+            ColoredLogger.log_server("Shutting down server...")
+            server_process.terminate()
+            server_process.wait(timeout=THREAD_CLEANUP_TIMEOUT_S)
+
+        stop_event.set()
+        # Wait for all threads to complete
+        if "server_thread" in locals():
+            server_thread.join(timeout=THREAD_CLEANUP_TIMEOUT_S)
+        if "server_error_thread" in locals():
+            server_error_thread.join(timeout=THREAD_CLEANUP_TIMEOUT_S)
+
+        # Wait some time to make sure everyting is cleanend up.
+        time.sleep(5)
+
+    return results
+
+
+def upload_results_to_s3(s3_path, results, service_metadata):
+    if any(result is None for result in results):
+        raise ValueError(
+            "Found None results during benchmarking. " "This should not have happened."
+        )
+
+    data_to_write = [{**result, **service_metadata} for result in results]
+    write_to_s3(data_to_write, s3_path)
+
+
+def main(pargs):
+    llm_config = get_llm_config(pargs.llm_config)
+    vllm_cli_args = get_vllm_cli_args(llm_config)
+
+    results = run_vllm_benchmark(vllm_cli_args)
+
+    # "A10" without "G" to match existing dashboard tags
+    accelerator = (
+        "A10"
+        if llm_config["accelerator_type"] == "A10G"
+        else llm_config["accelerator_type"]
+    )
+    tag = f"{accelerator}-TP{llm_config['engine_kwargs']['tensor_parallel_size']}"
+    service_metadata = {
+        "cloud_name": "",
+        "service_name": "",
+        "py_version": pargs.py_version,
```

Comment:
can you simply retrive py_version locally in this script instead of passing it in?